### PR TITLE
Pairtools sort

### PR DESCRIPTION
**Pull request by jianhong**
## PR checklist

- [ x ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ x ] If necessary, include test data in your PR.
- [ x ] Remove all TODO statements.
- [ x ] Emit the `<SOFTWARE>.version.txt` file.
- [ x ] Follow the naming conventions.
- [ x ] Follow the parameters requirements.
- [ x ] Follow the input/output options guidelines.
- [ x ] Add a resource `label`
- [ x ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [ x ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ x ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: software/pairtools/sort/main.nf [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    } else {
+        container "quay.io/biocontainers/pairtools:0.3.0--py37hb9c2fc3_5"
+    }
+
+    input:
+    tuple val(meta), path(input)
+
+    output:
+    tuple val(meta), path("*.pairs.gz"), emit: sorted
+    path "*.version.txt"                      , emit: version
```
Comment (drpatelh) [inline software/pairtools/sort/main.nf:26]:
```suggestion
    tuple val(meta), path("*.pairs.gz"), emit: sorted
    path "*.version.txt"               , emit: version
```

File: software/pairtools/sort/main.nf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    def software = getSoftwareName(task.process)
+    def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
+    def mem      = task.memory.toString().replaceAll(/(\s|\.|B)+/, '')
+    """
+    pairtools sort \\
+        $options.args \\
+        --nproc $task.cpus \\
+        --memory "$mem" \\
+        -o ${prefix}.pairs.gz \\
+        $input
```
Comment (drpatelh) [inline software/pairtools/sort/main.nf:38]:
```suggestion
    pairtools \\
    sort \\
        $options.args \\
        --nproc $task.cpus \\
        --memory "$mem" \\
        -o ${prefix}.pairs.gz \\
        $input
```

File: software/pairtools/sort/meta.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - version:
+      type: file
+      description: File containing software version
+      pattern: "*.{version.txt}"
+  - sorted:
+      type: file
+      description: Sorted pairs file
+      pattern: "*.{sorted.pairs.gz}"
```
Comment (drpatelh) [inline software/pairtools/sort/meta.yml:37]:
```suggestion
      pattern: "*.{pairs.gz}"
```

File: tests/software/pairtools/sort/test.yml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+- name: pairtools sort test_pairtools_sort
+  command: nextflow run tests/software/pairtools/sort -entry test_pairtools_sort -c tests/config/nextflow.config
+  tags:
+    - pairtools/sort
+    - pairtools
+  files:
+    - path: output/pairtools/testsorted.pairs.gz
```
Comment (drpatelh) [inline tests/software/pairtools/sort/test.yml:7]:
```suggestion
    - path: output/pairtools/test.sorted.pairs.gz
```

File: tests/software/pairtools/sort/main.nf [hunk 5]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env nextflow
+
+nextflow.enable.dsl = 2
+
+include { PAIRTOOLS_SORT } from '../../../../software/pairtools/sort/main.nf' addParams( options: ['suffix':'sorted'] )
```
Comment (drpatelh) [inline tests/software/pairtools/sort/main.nf:5]:
```suggestion
include { PAIRTOOLS_SORT } from '../../../../software/pairtools/sort/main.nf' addParams( options: ['suffix':'.sorted'] )
```

**Review by drpatelh — COMMENTED** [review]
💥

**Review by drpatelh — APPROVED** [review]
💥

**Comment by jianhong** [comment]
Thank you @drpatelh .

**Pull request merged**